zproject/backends.py

```diff
@@ -1547,6 +1549,30 @@ def social_auth_finish(
     else:
         is_signup = False
 
+    extra_attrs = return_data.get("extra_attrs", {})
+    if (
+        user_profile is not None
+        and extra_attrs
+        and realm.subdomain in settings.SOCIAL_AUTH_SYNC_CUSTOM_ATTRS_DICT
+    ):
+        # This is only supported for SAML right now, though the design
+        # is meant to be easy to extend this to other backends if desired.
+        assert backend.name == "saml"
+        custom_profile_field_name_to_attr_name = settings.SOCIAL_AUTH_SYNC_CUSTOM_ATTRS_DICT[
+            realm.subdomain
+        ].get(backend.name, {})
```

Comment:
Would it be cleaner to `attrs_to_sync = settings.SOCIAL_AUTH_SYNC_CUSTOM_ATTRS_DICT.get(realm.subdomain)` before the above if statement (when this is `and attrs_to_sync is not None`), to avoid accessing this data structure twice?

Reply:
Yeah, much better to extract that, done